main/ejml-dsparse/src/org/ejml/sparse/csc/misc/ImplCommonOpsWithSemiRing_DSCC.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2009-2020, Peter Abeles. All Rights Reserved.
+ *
+ * This file is part of Efficient Java Matrix Library (EJML).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ejml.sparse.csc.misc;
+
+import org.ejml.data.DGrowArray;
+import org.ejml.data.DMatrixSparseCSC;
+import org.ejml.data.IGrowArray;
+import org.ejml.ops.DSemiRing;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+
+import static org.ejml.UtilEjml.adjust;
+import static org.ejml.sparse.csc.mult.ImplSparseSparseMultWithSemiRing_DSCC.multAddColA;
+
+/**
+ * based on ImplCommonOps_DSCC
+ */
+public class ImplCommonOpsWithSemiRing_DSCC {
```

Comment:
Maybe this class should extend `ImplCommonOpsWithSemiRing`?

Reply:
What would be the value here?
I don't see how I can override any methods, as the signature differs regarding the semi-ring parameter.
I could overwrite the ones without semi-rings to use call the semi-ring version and call with the natural plus-times semi-ring

Reply:
I guess you could avoid the call to [tranpose](https://github.com/lessthanoptimal/ejml/blob/60093bc94a6b9577f08e69ef36371eec2ba4383d/main/ejml-dsparse/src/org/ejml/sparse/csc/misc/ImplCommonOps_DSCC.java#L47) since the signatures match, but probably it's not worth it. In general it's probably best to keep everything flat and use static methods to avoid confusion.

Reply:
I initially added the transpose method, as I wanted a common place for Graphblas operators.
However I also removed apply and reduce, as I didn't think having a redirect here is worth it.

I will remove the transpose function from this class, thank you for spotting this  :+1: